Source/Core/DolphinQt/Debugger/MemoryWidget.cpp

```diff
@@ -365,17 +381,45 @@ void MemoryWidget::ValidateSearchValue()
 {
   QFont font;
   QPalette palette;
+  m_data_preview->setText(tr(""));
 
   if (m_find_hex->isChecked() && !m_data_edit->text().isEmpty())
   {
     bool good;
-    m_data_edit->text().toULongLong(&good, 16);
+    u64 value = m_data_edit->text().toULongLong(&good, 16);
 
     if (!good)
     {
       font.setBold(true);
       palette.setColor(QPalette::Text, Qt::red);
     }
+
+    else
+    {
+      int data_length = m_data_edit->text().size();
+      QString hexstring;
+      if (data_length > 8)
+        hexstring = QStringLiteral("%1").arg(value, 16, 16, QLatin1Char('0')).toUpper();
+      else if ((m_type_u8->isChecked() || m_type_ascii->isChecked()) && data_length < 5)
+      {
+        if (data_length > 2)
+          hexstring = QStringLiteral("%1").arg(value, 4, 16, QLatin1Char('0')).toUpper();
+        else
+          hexstring = QStringLiteral("%1").arg(value, 2, 16, QLatin1Char('0')).toUpper();
+      }
+      else if (m_type_u16->isChecked() && data_length <= 4)
+      {
+        hexstring = QStringLiteral("%1").arg(value, 4, 16, QLatin1Char('0')).toUpper();
+      }
+      else
+      {
+        hexstring = QStringLiteral("%1").arg(value, 8, 16, QLatin1Char('0')).toUpper();
+      }
+      int hsize = hexstring.size();
+      for (int i = 2; i < hsize; i += 2)
+        hexstring.insert(hsize - i, tr(" "));
```

Comment:
this also doesn't need to be a translatable string.

Reply:
What do I replace this with? QStringLIteral? I'm a bit confused on the proper thing to use here, it seems different from examples on the net.

Reply:
Only a single character is being inserted so you can just use `QChar{' '}`

Reply:
err, my bad, I forgot we have constructors like that disabled, so you'd have to use `QLatin1Char{' '}` in this case.